lua_examples/lfs/lfs_fragments.lua

```diff
@@ -0,0 +1,124 @@
+  getfenv().LFS = setmetatable(lfs_t,lfs_t)
+
+  local function loader_flash(module)
+    local fn, ba = index(module)
+    return ba and "Module not in LFS" or fn 
+  end
+  package.loaders[3] = loader_flash
+
+end
+
+-----------------------------------------------------------------------------------
+--
+-- File: init.lua
+--
+-- With the previous example you still need an init.lua to bootstrap the _init 
+-- module in LFS.  Here is an example.  It's a good idea either to use a timer 
+-- delay or a GPIO pin during development, so that you as developer can break into 
+-- the boot sequence if there is a problem with the _init bootstrap that is causing
+-- a panic loop.  Here is one example of how you might do this.  You have a second to
+-- inject tmr.stop(0) into UART0.  Extend if your reactions can't meet this.
+--
+-- You also want to do autoload the LFS, for example by adding the following:
+--
+if node.flashindex() == nil then 
+  node.flashreload('flash.img') 
+end
+
+tmr.alarm(0, 1000, tmr.ALARM_SINGLE, 
+  function()
+    local fi=node.flashindex; return pcall(fi and fi'_init')
+  end)
+
+-----------------------------------------------------------------------------------
+--
+-- The debug.getstrings function can be used to get a listing of the RAM (or ROM
+-- if LFS is loaded), as per the following example, so you can do this at the 
+-- interactive prompt or call it as a debug function during a running application. 
+--
+do
+  local a=debug.getstrings'RAM'
+  for i =1, #a do a[i] = ('%q'):format(a[i]) end
+  print ('local preload='..table.concat(a,','))
+end
+
+-----------------------------------------------------------------------------------
+--
+-- File: LFS_dummy_strings.lua
+--
+-- luac.cross -f will generate a ROM string table that includes all strings 
+-- referenced in the loaded modules.  If you want to preload other string constants
+-- hen the trick is to include a dummy module in the LFS.  You never need to call
+-- this.  It's inclusion is enough to add the strings to the ROM table.  Once in
+-- the ROM table, then you can use them in your application without incuring any 
+-- RAM or Lua Garbage Collector (LGC) overhead.  Here is a useful starting point,
+-- but you can add to this for your application.  
+--
+-- The trick is to build the LFS as normal then run the previous example from your
+-- running application and append these lines to this file.
+--
+local preload = "?.lc;?.lua", "@init.lua", "_G", "_LOADED", "_LOADLIB", "__add", 
+"__call", "__concat", "__div", "__eq", "__gc", "__index", "__le", "__len", "__lt", 
+"__mod", "__mode", "__mul", "__newindex", "__pow", "__sub", "__tostring", "__unm", 
+"collectgarbage", "cpath", "debug", "file", "file.obj", "file.vol", "flash", 
+"getstrings", "index", "ipairs", "list", "loaded", "loader", "loaders", "loadlib", 
+"module", "net.tcpserver", "net.tcpsocket", "net.udpsocket", "newproxy", "package", 
+"pairs", "path", "preload", "reload", "require", "seeall", "wdclr"
+
```

Comment:
Okay @TerryE, this file is amazingly good and useful. Can I suggest you mention it in the API docs for `node.flashindex()`?

Reply:
I am rewriting the LFS paper to make it more orientated to the Lua app developer.  I have covered this in there.  My intention is to hoist this paper or some derivative into the formal documentation set.